libraries/standard/mqtt/src/iot_mqtt_operation.c

```diff
@@ -436,6 +463,107 @@ static bool _completePendingSend( _mqttOperation_t * pOperation,
 
 /*-----------------------------------------------------------*/
 
+static IotMqttError_t _initializeOperation( _mqttConnection_t * pMqttConnection,
+                                            _mqttOperation_t * pOperation,
+                                            uint32_t flags,
+                                            const IotMqttCallbackInfo_t * pCallbackInfo )
+{
+    IotMqttError_t status = IOT_MQTT_SUCCESS;
+    bool waitable = ( ( flags & IOT_MQTT_FLAG_WAITABLE ) == IOT_MQTT_FLAG_WAITABLE );
+
+    IotMqtt_Assert( pMqttConnection != NULL );
+    IotMqtt_Assert( pOperation != NULL );
+   
+    /* Clear the operation data. */
+    ( void ) memset( pOperation, 0x00, sizeof( _mqttOperation_t ) );
+
+    /* Initialize some members of the new operation. */
+    pOperation->pMqttConnection = pMqttConnection;
+    pOperation->u.operation.jobReference = 1;
+    pOperation->u.operation.flags = flags;
+    pOperation->u.operation.status = IOT_MQTT_STATUS_PENDING;
+
+    /* Check if the waitable flag is set. If it is, create a semaphore to
+     * wait on. */
+    if( waitable == true )
+    {
+        /* Create a semaphore to wait on for a waitable operation. */
+        if( IotSemaphore_Create( &( pOperation->u.operation.notify.waitSemaphore ), 0, 1 ) == false )
+        {
+            IotLogError( "(MQTT connection %p) Failed to create semaphore for "
+                         "waitable operation.",
+                         pMqttConnection );
+
+            status = IOT_MQTT_NO_MEMORY;
+        }
+        else
+        {
+            /* A waitable operation is created with an additional reference for the
+             * Wait function. */
+            ( pOperation->u.operation.jobReference )++;
+        }
+    }
+    else
+    {
+        /* If the waitable flag isn't set but a callback is, copy the callback
+         * information. */
+        if( pCallbackInfo != NULL )
+        {
+            pOperation->u.operation.notify.callback = *pCallbackInfo;
+        }
+    }
+
+    return status;
+}
+
+/*-----------------------------------------------------------*/
+
+static bool _sendPingRequest( _mqttConnection_t * pMqttConnection,
+                              _mqttOperation_t * pPingreqOperation )
+{
+    size_t bytesSent = 0;
+    bool status = true;
+    uint32_t swapStatus = 0;
```

Comment:
Since they're here, `bytesSent` and `swapStatus` are no longer used in `_IotMqtt_ProcessKeepAlive` and can be deleted from lines 808 and 810